Modules/atexitmodule.c

```diff
@@ -191,6 +196,29 @@ atexit_register(PyObject *module, PyObject *args, PyObject *kwargs)
     }
 
     struct atexit_state *state = get_atexit_state();
+    if (_Py_atomic_load_int(&state->handler_calling_started)) {
```

Comment:
arguably all of the logic around blocking new atexit.registers from happening could be left out of this PR.  it felt relevant though.  we never really defined or documented what happened before (registrations made from a handler were never called). This makes it explicit and gives some indication as to when a registration is being ignored.